google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -196,11 +206,24 @@ public DocumentReference document(@Nonnull String documentPath) {
 
     ApiStreamObserver<BatchGetDocumentsResponse> responseObserver =
         new ApiStreamObserver<BatchGetDocumentsResponse>() {
+          int numResponses;
+
           @Override
           public void onNext(BatchGetDocumentsResponse response) {
             DocumentReference documentReference;
             DocumentSnapshot documentSnapshot;
 
+            numResponses++;
+            if (numResponses == 1) {
+              Tracing.getTracer()
+                  .getCurrentSpan()
```

Comment:
I couldn't determine if the context is propagated to this point from the "runTransaction". Is the onNext executed in the same thread as runTransaction?